src/test/scala/com/microsoft/hyperspace/index/RefreshIndexTests.scala

```diff
@@ -0,0 +1,136 @@
+    // save test data non-partitioned.
+    SampleData.save(
+      spark,
+      nonPartitionedDataPath,
+      Seq("Date", "RGUID", "Query", "imprs", "clicks"))
+    val nonPartitionedDataDF = spark.read.parquet(nonPartitionedDataPath)
+
+    // save test data partitioned.
+    SampleData.save(
+      spark,
+      partitionedDataPath,
+      Seq("Date", "RGUID", "Query", "imprs", "clicks"),
+      Some(Seq("Date", "Query")))
+    val partitionedDataDF = spark.read.parquet(partitionedDataPath)
+
+    Seq(nonPartitionedDataPath, partitionedDataPath).foreach { loc =>
+      withSQLConf(
+        IndexConstants.INDEX_LINEAGE_ENABLED -> "true",
+        IndexConstants.REFRESH_DELETE_ENABLED -> "true") {
+        withIndex(indexConfig.indexName) {
+          val dfToIndex =
+            if (loc.equals(nonPartitionedDataPath)) nonPartitionedDataDF else partitionedDataDF
+          hyperspace.createIndex(dfToIndex, indexConfig)
+
+          // delete some source data file.
+          val dataPath =
+            if (loc.equals(nonPartitionedDataPath)) new Path(nonPartitionedDataPath, "*parquet")
+            else new Path(partitionedDataPath + "/Date=2018-09-03/Query=ibraco", "*parquet")
+
+          val dataFileNames = dataPath
+            .getFileSystem(new Configuration)
+            .globStatus(dataPath)
+            .map(_.getPath)
+
+          assert(dataFileNames.length > 0)
+          val fileToDelete = dataFileNames.head
+          FileUtils.delete(fileToDelete)
+
+          val originalIndexDF = spark.read.parquet(s"$systemPath/${indexConfig.indexName}/" +
+            s"${IndexConstants.INDEX_VERSION_DIRECTORY_PREFIX}=0")
+          val delCount =
+            originalIndexDF
+              .filter(s"""${IndexConstants.DATA_FILE_NAME_COLUMN} ==  "$fileToDelete"""")
+              .count()
+
+          hyperspace.refreshIndex(indexConfig.indexName)
+
+          val refreshedIndexDF = spark.read.parquet(s"$systemPath/${indexConfig.indexName}/" +
+            s"${IndexConstants.INDEX_VERSION_DIRECTORY_PREFIX}=1")
+
+          // validate only index records whose lineage is the deleted file are removed.
+          assert(refreshedIndexDF.count() == (originalIndexDF.count() - delCount))
+
+          val lineageFileNames = refreshedIndexDF
+            .select(IndexConstants.DATA_FILE_NAME_COLUMN)
+            .distinct()
+            .collect()
+            .map(r => new Path(r.getString(0)))
+
+          assert(!lineageFileNames.contains(fileToDelete))
+        }
+      }
+    }
+  }
+
+  test("Validate refresh delete fails as expected on an index without lineage.") {
+    SampleData.save(
+      spark,
+      nonPartitionedDataPath,
+      Seq("Date", "RGUID", "Query", "imprs", "clicks"))
+    val nonPartitionedDataDF = spark.read.parquet(nonPartitionedDataPath)
+
+    withSQLConf(
+      IndexConstants.INDEX_LINEAGE_ENABLED -> "false",
+      IndexConstants.REFRESH_DELETE_ENABLED -> "true") {
```

Comment:
What is the expected behavior when both are set to false and some data gets deleted from the underlying source?

Reply:
That is the default index refresh behavior we have today: Ignoring existing index files and fully building index from scratch completely from the source files. Default values of config flags makes such a behavior the default behavior today.